splice_si_api/src/main/java/com/splicemachine/si/impl/server/SITransactor.java

```diff
@@ -325,12 +327,12 @@ private void releaseLocksForKvBatch(Pair<KVPair, Lock>[] locks){
             conflictingChildren[i] = conflictResults.getChildConflicts();
             boolean addFirstOccurrenceToken = false;
 
-            if (!skipConflictDetection) {
+            if (checkedConflicts) {
```

Comment:
Can you add a comment there explaining the defensive approach, as well a reference to a jira that tracks our efforts relaxing that defensive approach in the interest of performance?
Also, make sure that PhysicalDeletionIt still pass after this change.

Reply:
done